EIPS/eip-3540.md

```diff
@@ -0,0 +1,218 @@
+| format      | 1-byte  | 0xEF  | |
+| magic       | n-byte(s)  | TBD   | n >= 0 (zero in the best case) |
+| version     | 1-byte | 0x01  | means EOF1 |
+
+This is followed by one or more section headers:
+
+| description  | length  | | 
+|--------------|---------|-|
+| section_kind | 1-byte  | Encoded as a 8-bit unsigned number. |
+| section_size | 2-bytes | Encoded as a 16-bit unsigned big-endian number. |
+
+The section kinds are defined as follows:
+
+| section_kind | meaning    |
+|--------------|------------|
+| 0            | terminator |
+| 1            | code       |
+| 2            | data       |
+
+If the terminator is encountered, section size MUST NOT follow.
+
+The section contents follow after the header, in the order and size they are defined, without any padding bytes.
+
+To summarise, the bytecode has the following format:
+```
+format, magic, version, (section_kind, section_size)+, 0, <section contents>
+```
+
+#### Validation rules
+
+A bytestream starting with the *EOF prefix* declares itself conforming to the rules according to its version.
+
+1. The rules of `version=1` are specified below:
+- `section_size` MUST NOT be 0.
+- Exactly one code section MUST be present.
+- The code section MUST be the first section.
+- A single data section MAY follow the code section.
+- Stray bytes outside of sections MUST NOT be present. This includes trailing bytes after the last section.
+2. Any other version is invalid.
+
+(*Note:* Contract creation code SHOULD set the section size of the data section so that the constructor arguments fit it.)
+
+#### Changes to execution semantics
+
+For clarity, the *container* refers to the complete account code, while *code* refers to the contents of the code section only.
+
+1. Jumpdest analysis is only run on the *code*.
+2. Execution starts at the first byte of the *code*, and `PC` is set to this position within the container format (e.g. `PC=10` for a *container* with a code and data section).
+3. If `PC` goes outside of the code section bounds, execution  aborts with failure.
+4. `PC` returns the current position within the *container*.
+5. `JUMP`/`JUMPI` uses an absolute offset within the *container*.
+6. `CODECOPY`/`CODESIZE`/`EXTCODECOPY`/`EXTCODESIZE`/`EXTCODEHASH` keeps operating on the entire *container*.
+7. The input to `CREATE`/`CREATE2` is still the entire *container*.
+
+#### Changes to contract creation semantics
+
+For clarity, the *EOF prefix* together with a version number $v$ is denoted as the *EOF$v$ prefix*, e.g. *EOF1 prefix*. 
+
+1. If _initcode's container_ has EOF1 prefix it must be valid EOF1 code.
+2. If _code's container_ has EOF1 prefix it must be valid EOF1 code.
+
+## Rationale
+
+EVM and/or account versioning has been discussed numerous times over the past years. This proposal aims to learn from them. A good starting point collecting the previous proposals is [this thread](https://ethereum-magicians.org/t/ethereum-account-versioning/3508).
```

Comment:
Don't include external links in EIPs.  Remember, EIPs are technical specifications.  For complex topics (like this one), you **SHOULD** have a centralized external document for linking to discussions, further reading, etc. (similar to EIP-1559).

Also, this statement doesn't really belong in the rationale section as it doesn't actually explain any particular decisions.  Consider moving the sentiment to the motivation section, or just removing it (shorter EIPs tend to be better).
```suggestion
```

Reply:
How come a lot of EIPs can have external links in the abstract and rationale? Two recent examples are: https://eips.ethereum.org/EIPS/eip-2938 (starts with a very similar FEM link in the abstract) and https://eips.ethereum.org/EIPS/eip-3529 (has links to various draft proposals).